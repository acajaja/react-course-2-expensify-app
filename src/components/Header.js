import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = (props) => (
    <header>
        <h1>Expensify</h1>
        <nav>
            <NavLink to="/" activeClassName="is-active" title="Home" exact={true}>Home</NavLink>
            <NavLink to="/create" activeClassName="is-active" title="Create">Create</NavLink>
            <NavLink to="/edit" activeClassName="is-active" title="Edit">Edit</NavLink>
            <NavLink to="/help" activeClassName="is-active" title="Help">Help</NavLink>
        </nav>
    </header>
);

export default Header;
