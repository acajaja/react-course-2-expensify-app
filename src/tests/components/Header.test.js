import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Header from '../../components/Header';

test('Test rendering header', () => {
    const wrapper = shallow(<Header />);
    expect(toJSON(wrapper)).toMatchSnapshot();

    // Enzyme shallow wrapper
    // expect(wrapper.find('h1').text()).toBe('Expensify');
    // React shallow renderer
    // const renderer = new ReactShallowRenderer();
    // renderer.render(<Header />);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
});
