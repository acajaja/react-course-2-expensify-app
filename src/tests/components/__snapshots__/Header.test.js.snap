// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test rendering header 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <nav>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      exact={true}
      title="Home"
      to="/"
    >
      Home
    </NavLink>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      title="Create"
      to="/create"
    >
      Create
    </NavLink>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      title="Edit"
      to="/edit"
    >
      Edit
    </NavLink>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      title="Help"
      to="/help"
    >
      Help
    </NavLink>
  </nav>
</header>
`;
